packages/techdocs-common/src/stages/generate/techdocs.ts

```diff
@@ -66,24 +66,36 @@ export class TechdocsGenerator implements GeneratorBase {
   public async run({
     directory,
     dockerClient,
+    expectedResultDir,
     parsedLocationAnnotation,
   }: GeneratorRunOptions): Promise<GeneratorRunResult> {
-    const tmpdirPath = os.tmpdir();
-    // Fixes a problem with macOS returning a path that is a symlink
-    const tmpdirResolvedPath = fs.realpathSync(tmpdirPath);
-    const resultDir = fs.mkdtempSync(
-      path.join(tmpdirResolvedPath, 'techdocs-tmp-'),
-    );
+    // If provided with a directory to output the generated files in, use that directory.
+    // Else create a new temporary directory and return path.
```

Comment:
I know that this helps to not break the interface, but it also risks piling up legacy :) It could be argued that the caller should be required to make a temp folder (and also clean it up, symmetrically) if what they wanted is a temp folder.

Reply:
I think you are right. This should be `generate(what, where)`